src/auth/authenticator/oicd/jwt.go

```diff
@@ -0,0 +1,82 @@
+package oicd
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/golang-jwt/jwt"
+)
+
+type JWTChecker struct {
+	certs     []*x509.Certificate
+	parser    *jwt.Parser
+	claimsCfg *ClaimsConfig
+}
+
+func NewJWTChecker(certs []*x509.Certificate, claimsCfg *ClaimsConfig, skipClaimsValidation bool) *JWTChecker {
+	return &JWTChecker{
+		certs:     certs,
+		claimsCfg: claimsCfg,
+		parser: &jwt.Parser{
+			SkipClaimsValidation: skipClaimsValidation,
+		},
+	}
+}
+
+func (checker *JWTChecker) Check(_ context.Context, bearerToken string) (*Claims, error) {
+	if len(checker.certs) == 0 {
+		// If no certificate provided we deactivate authentication
+		return nil, nil
+	}
+
+	// Parse and validate token injected in context
+	token, err := checker.parser.ParseWithClaims(
+		bearerToken,
+		&Claims{cfg: checker.claimsCfg},
+		checker.keyFunc,
+	)
+	if err != nil {
+		return nil, err
+	} else if !token.Valid {
+		return nil, fmt.Errorf("invalid access token")
```

Comment:
I think that no `fmt.Errorf` should exist under `src` folder. This seems to be an Unauthorized error